src/ReactHardwareComponent.js

```diff
@@ -42,16 +64,23 @@ type ViewConfigPropTypes = {
  * @constructor ReactHardwareComponent
  * @extends ReactComponent
  * @extends ReactMultiChild
- * @param {!object} viewConfig View Configuration.
  */
-const ReactHardwareComponent = function(
-  element: React$Element
-) {
-  this.viewConfig = DEFAULT_VIEW_CONFIG;
+const ReactHardwareComponent = function(element: React$Element) {
   this._rootNodeID = null;
   this._renderedChildren = null;
   this._io = null;
-  this.construct(element);
+  this._currentElement = element;
+  this.viewConfig = DEFAULT_VIEW_CONFIG;
+  if (typeof element.type === 'string') {
+    this.viewConfig = getViewConfig(element.type);
+  }
+  if (process.env.NODE_ENV !== 'production') {
+    invariant(
+      element.type === 'pin' || element.type === 'container',
+      '%s is not a defined component',
```

Comment:
`Attempted to render an unsupported generic component "%s". Must be "pin" or "container". Check the render method of %s`

You can see how to do this at https://github.com/facebook/react/blob/eb85b7b2c989faeb6ed8cd787cf67bd5f6d3ec7d/src/renderers/dom/client/ReactMount.js#L319-L321